Disable qvitter-delete-notice if ActivityModeration plugin exists

mmn opened the pull request:
I think this won't cause any problems with older installations. If the GNU social instance is newer and has the ActivityModeration plugin, remote notice deletion is handled through that instead of Qvitter.

mmn commented:
Hm, this doesn't seem to hide the delete verb notices though, so there are items like these listed: "<Mikael & Nordfeldth> deleted notice {{tag:social.umeahackerspace.se,2015-10-03:noticeId=280252:objectType=note}}."

hannesmannerheim commented:
i'll check it out

hannesmannerheim commented:
i updated from nightly, ran upgrade.php and added to config:

´
addPlugin('ActivityVerb');
addPlugin('ActivityModeration');
´
but i can't delete notices anymore. i click delete, confirm delete and then i get:
<img width="547" alt="skarmavbild 2015-10-04 kl 13 03 33" src="https://cloud.githubusercontent.com/assets/2437812/10267537/a4808ff2-6a98-11e5-8a6e-ed5860a7f960.png">

and if i click yes i get
<img width="537" alt="skarmavbild 2015-10-04 kl 13 03 47" src="https://cloud.githubusercontent.com/assets/2437812/10267539/abe0ad22-6a98-11e5-9425-abe46a00d4a5.png">

mmn commented:
On 2015-10-04 13:06, Hannes Mannerheim wrote:
> i updated from nightly, ran upgrade.php and added to config:
> 
> ´
> addPlugin('ActivityVerb');
> addPlugin('ActivityModeration');

You shouldn't have to add those plugins manually, they are listed in
lib/default.php in the $config['plugins']['core'] array.

> but i can't delete notices anymore. i click delete, confirm delete and
> then i get:
> skarmavbild 2015-10-04 kl 13 03 33

Oh, interesting. Notice_location is in db/core.php so it should be
checked and fixed when running upgrade.php

Do you have a notice_location table in that site's database?

-- 
Mikael Nordfeldth
XMPP/mail: mmn@hethane.se
OpenPGP fpr: AE68 9813 0B7C FCE3 B2FA  727B C7CE 635B B52E 9B31

hannesmannerheim commented:
oh, ok, i thought it was an opt-in feature?

no, no notice_location table. maybe upgrade.php failed silently

mmn commented:
On 2015-10-04 13:40, Hannes Mannerheim wrote:
> oh, ok, i thought it was an opt-in feature?
> 
> no, no notice_location table. maybe upgrade.php failed silently

I don't think it can fail silently :)

Does scripts/upgrade.php say anything about "moving location data"?

Or maybe an even better question, if you in the mysql/mariadb CLI do:

DESCRIBE notice;

does it list columns like 'lat', 'lon, 'location_id' or 'location_ns'?

In that case, Notice::beforeSchemaUpdate() doesn't get run. Can you make
sure that code is in Notice.php and have another look at the output of
the upgrade script?

-- 
Mikael Nordfeldth
XMPP/mail: mmn@hethane.se
OpenPGP fpr: AE68 9813 0B7C FCE3 B2FA  727B C7CE 635B B52E 9B31

hannesmannerheim commented:
there were errors in the log from upgrade.php, outdated extensions...... trying to fix it. but my consoleophobia is not helping

hannesmannerheim commented:
finally got the upgrade script working again! now deleting of notices works :) i'll check why the activity notices are not hidden

hannesmannerheim commented:
looks like the activity notice produced is saved with object type ``http://activitystrea.ms/schema/1.0/note``

shoudn't it be ``http://activitystrea.ms/schema/1.0/activity`` ?

hannesmannerheim commented:
i made qvitter consider the delete-"note":s to be activity notices (that are hidden)
